src/compiler/scala/tools/nsc/typechecker/SyntheticMethods.scala

```diff
@@ -359,13 +360,14 @@ trait SyntheticMethods extends ast.TreeDSL {
         for ((m, impl) <- methods ; if shouldGenerate(m)) yield impl()
       }
       def extras = {
-        if (needsReadResolve) {
+        if (needsModuleSerializationProxy) {
```

Comment:
I do wonder if we should only generate this method if we are generating java class files. This is not required for js and native backends is it?